src/util/BitString.ts

```diff
@@ -0,0 +1,43 @@
+import { HexConverter } from './HexConverter.js';
+import { DataHash } from '../hash/DataHash.js';
+
+export class BitString {
+  /**
+   * Represents a bit string as a bigint.
+   * The first bit is always set to 1 to not lose leading bits when converting to bigint.
```

Comment:
put this comment on top if the class? also, this is a classic point of ambiguity. Let's be very specific:

Bit string (big-endian notation) with most significant bit (leftmost) always set to 1 to preserve leading zeroes